contrib/IECoreArnold/src/IECoreArnold/MeshAlgo.cpp

```diff
@@ -84,121 +84,44 @@ const T *variableData( const PrimitiveVariableMap &variables, const std::string
-		AiNodeSetArray( node, (setName + "idxs").c_str(), indicesArray );
-	}
-
-	variables.erase( sName );
-	variables.erase( tName );
-	variables.erase( indicesName );
-}
-
-// Version of the above for when we have no indices available.
-void convertUVSet( const std::string &setName, PrimitiveVariableMap &variables, const vector<int> &vertexIds, AtNode *node )
+void convertUVSet( PrimitiveVariableMap::const_iterator uvIt, const vector<int> &vertexIds, AtNode *node )
 {
-	const std::string sName = setName == "" ? "s" : setName + "_s";
-	const std::string tName = setName == "" ? "t" : setName + "_t";
-
-	const FloatVectorData *sData = variableData<FloatVectorData>( variables, sName );
-	const FloatVectorData *tData = variableData<FloatVectorData>( variables, tName );
-
-	if( !( sData && tData ) )
-	{
-		return;
-	}
+	const V2fVectorData *uvData = runTimeCast<V2fVectorData>( uvIt->second.data.get() );
 
-	PrimitiveVariable::Interpolation sInterpolation = variables.find( "s" )->second.interpolation;
-	PrimitiveVariable::Interpolation tInterpolation = variables.find( "t" )->second.interpolation;
-	if( sInterpolation != tInterpolation )
+	if( !uvData )
 	{
-		msg(
-			Msg::Warning, "ToArnoldMeshConverter::doConversion",
-			boost::format( "Variables \"%s\" and \"%s\" have different interpolation - not generating uvs." ) % sName % tName
-		);
 		return;
 	}
 
-	if( sInterpolation != PrimitiveVariable::Varying && sInterpolation != PrimitiveVariable::Vertex && sInterpolation != PrimitiveVariable::FaceVarying )
+	if( uvIt->second.interpolation != PrimitiveVariable::Varying && uvIt->second.interpolation != PrimitiveVariable::Vertex && uvIt->second.interpolation != PrimitiveVariable::FaceVarying )
 	{
 		msg(
 			Msg::Warning, "ToArnoldMeshConverter::doConversion",
-			boost::format( "Variables \"%s\" and \"%s\" have different interpolation types - not generating uvs." ) % sName % tName
+			boost::format( "Variable \"%s\" has an invalid interpolation type - not generating uvs." ) % uvIt->first
 		);
 		return;
 	}
 
-	const vector<float> &s = sData->readable();
-	const vector<float> &t = tData->readable();
+	const vector<Imath::V2f> &uvs = uvData->readable();
 
-	AtArray *uvsArray = AiArrayAllocate( s.size(), 1, AI_TYPE_POINT2 );
-	for( size_t i = 0, e = s.size(); i < e; ++i )
+	AtArray *uvsArray = AiArrayAllocate( uvs.size(), 1, AI_TYPE_POINT2 );
+	for( size_t i = 0, e = uvs.size(); i < e; ++i )
 	{
-		AtPoint2 uv = { s[i], t[i] };
+		AtPoint2 uv = { uvs[i][0], uvs[i][1] };
 		AiArraySetPnt2( uvsArray, i, uv );
 	}
 
-	AtArray *indicesArray = NULL;
-	if( sInterpolation == PrimitiveVariable::FaceVarying )
+	AtArray *indicesArray = nullptr;
+	if( uvIt->second.indices )
+	{
+		const vector<int> &indices = uvIt->second.indices->readable();
+		indicesArray = AiArrayAllocate( indices.size(), 1, AI_TYPE_UINT );
+		for( size_t i = 0, e = indices.size(); i < e; ++i )
+		{
+			AiArraySetUInt( indicesArray, i, indices[i] );
```

Comment:
Seems to be another vote for unsigned int for indices?

Reply:
Pretty sure this code would be just as ugly though, unless there's some bulk AiArrayUInt thingy?

Reply:
Yeah, there's an `AiArrayConvert()` we could use.